homeassistant/components/ping/binary_sensor.py

```diff
@@ -131,32 +131,70 @@ def __init__(self, host, count):
                 self._ip_address,
             ]
 
-    def ping(self):
+    async def async_ping(self):
         """Send ICMP echo request and return details if success."""
-        pinger = subprocess.Popen(
-            self._ping_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        pinger = await asyncio.create_subprocess_exec(
```

Comment:
Spawning a subprocess with inactive current child watcher raises `RuntimeError`.

We always have a child watcher so that shouldn't ever been an issue.